src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -67,6 +76,58 @@ export type ReactNodeViewConstructor = (
   ...args: Parameters<NodeViewConstructor>
 ) => ReactNodeView;
 
+const REACT_PM_KEY = "reactpmkey";
+
+function findNearestRegistryKey(editorView: EditorView, pos: number) {
+  const parentElement = editorView.domAtPos(pos, 0).node as HTMLElement;
+  const nearestElementWithKey = parentElement?.closest(
+    `[data-${REACT_PM_KEY}]`
+  ) as HTMLElement | null;
```

Comment:
What is this if you don't cast it?

Reply:
Oh, maybe `Element`?

Reply:
Yeah, domAtPos returns a `Node` and closest returns an `Element`, neither of which have the APIs we need here. We could use type guards instead of casting, but... I'm pretty confident that these will _always_ be HTMLElements

Reply:
Alternatively, we could actually check `instanceof HTMLElement` and fall back to the root key in the event that `domAtPos` is not an `HTMLElement` or we can't find a closest ancestor with that property that is?

I've wondered what would happen if you tried to put an `<svg>` node in a ProseMirror document and have a `<text>` descendant of it. Would it be editable? These would be `SVGElement`. Interestingly, `SVGElement` also has `dataset`.